### PR TITLE
[ Chore ⚙ ] Added TS version for dailibuzz

### PR DESCRIPTION
**Pull request by zerothebahdman**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  nanoid      String?   @db.VarChar(255)
+  name        String    @db.VarChar(255)
+  image       String?   @db.VarChar(255)
+  url         String?   @db.VarChar(255)
+  source      String?   @db.VarChar(255)
+  category_id Int?
+  created_at  DateTime  @default(now()) @db.DateTime(0)
+  updated_at  DateTime  @db.DateTime(0)
+  expires_at  DateTime  @updatedAt @db.DateTime(0)
+  Category    Category? @relation(fields: [category_id], references: [id], onDelete: Cascade, onUpdate: Cascade, map: "articles_ibfk_1")
```
Comment (zerothebahdman) [inline prisma/schema.prisma:21]: Setup relationship between articles and categories

File: src/scrapper/class/PunchClass.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        .get();
+      return Promise.all(articles);
+    } catch (err: any) {
+      throw new Error(err.message);
+    }
+  }
+
+  public async exportResults(results: Article[]) {
+    // const getCategory = await category.findFirst({
+    //   where: { name: this.category },
```
Comment (codewithdiv) [inline src/scrapper/class/PunchClass.ts:74]: Using the await key word when creating an article in a for loop throws an error.

File: src/scrapper/class/PunchClass.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      //       // @ts-ignore
+      //       created_at: Date.now(),
+      //       // @ts-ignore
+      //       updated_at: Date.now(),
+      //       // @ts-ignore
+      //       expires_at: Date.now() + 7 * 24 * 60 * 60 * 1000,
+      //     },
+      //   });
+      //   log.info(`Completly saved to database`);
+      // }
```
Comment (zerothebahdman) [inline src/scrapper/class/PunchClass.ts]: looped through the results object and stored the resulting data in a data array then used that array of data in the createMany function to store articles to the database.

File: src/scrapper/class/PunchClass.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          name: i.articleTitle,
+          url: i.articleUrl,
+          nanoid: nanoid(),
+          image: i.articleImage,
+          category_id: getCategory[0].id,
+          source: this.source,
+        });
+      }
+      log.info(data);
+      await article.createMany({ data });
```
Comment (zerothebahdman) [inline src/scrapper/class/PunchClass.ts:100]: looped through the results object and stored the resulting data in a data array then used that array of data in the createMany function to store articles to the database.

File: src/controller/category.controller.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import CategoryService from '../services/category.service';
+
+export const createCategory = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { name } = req.body;
+    const category = await CategoryService.create(name);
```
Comment (zerothebahdman) [inline src/controller/category.controller.ts:13]: Asyncronously wait for the servie to return a valid data before sending back a response.

**Review by zerothebahdman — COMMENTED** [review]
Completed

**Pull request closed without merging**